types/firestore.d.ts

```diff
@@ -411,6 +410,139 @@ declare namespace FirebaseFirestore {
+     * @param {SetOptions=} options An object to configure the set behavior.
+     * @param {boolean=} options.merge - If true, set() merges the values
+     * specified in its data argument. Fields omitted from this set() call remain
+     * untouched.
+     * @param {Array.<string|FieldPath>=} options.mergeFields - If provided, set()
+     * only replaces the specified field paths. Any field path that is not
+     * specified is ignored and remains untouched.
+     * @returns {Promise<WriteResult>} A promise that resolves with the result of
+     * the write. Throws an error if the write fails.
+     */
+    set(
+      documentRef: DocumentReference,
+      data: DocumentData,
+      options?: SetOptions
+    ): Promise<WriteResult>;
+
+    /**
+     * Update fields of the document referred to by the provided
+     * [DocumentReference]{@link DocumentReference}. If the document doesn't yet
+     * exist, the update fails and the entire batch will be rejected.
+     *
+     * The update() method accepts either an object with field paths encoded as
+     * keys and field values encoded as values, or a variable number of arguments
+     * that alternate between field paths and field values. Nested fields can be
+     * updated by providing dot-separated field path strings or by providing
+     * FieldPath objects.
+     *
+     *
+     * A Precondition restricting this update can be specified as the last
+     * argument.
+     *
+     * @param {DocumentReference} documentRef A reference to the document to be
+     * updated.
+     * @param {UpdateData|string|FieldPath} dataOrField An object containing the
+     * fields and values with which to update the document or the path of the
+     * first field to update.
+     * @param {...(Precondition|*|string|FieldPath)} preconditionOrValues - An
+     * alternating list of field paths and values to update or a Precondition to
+     * restrict this update
+     * @returns {Promise<WriteResult>} A promise that resolves with the result of
+     * the write. Throws an error if the write fails.
+     */
+    update(
+      documentRef: DocumentReference,
+      dataOrField: UpdateData | string | FieldPath,
+      ...preconditionOrValues: Array<
+        {lastUpdateTime?: Timestamp} | unknown | string | FieldPath
+      >
+    ): Promise<WriteResult>;
+
+    /**
+     * Commits all writes that have been enqueued up to this point in parallel.
+     *
+     * Returns a Promise that resolves when all writes have been committed.
+     * The Promise will never be rejected.
+     *
+     * The Promise resolves immediately if there are no pending writes. Otherwise,
+     * the Promise waits for all previously issued writes, but it does not wait
+     * for writes that were added after the method is called. If you want to wait
+     * for additional writes, call `flush()` again.
+     *
+     * @return {Promise<void>} A promise that resolves when all writes have
```

Comment:
Please sync these docs with the documentation in the code when completed.

Reply:
synced.